tests/zfs-tests/tests/functional/suid/Makefile.am

```diff
@@ -0,0 +1,14 @@
+include $(top_srcdir)/config/Rules.am
+
+pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/suid
+
+dist_pkgdata_SCRIPTS = \
+	cleanup.ksh \
+	setup.ksh
+
+pkgexecdir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/suid
+
+pkgexec_PROGRAMS = suid_write_to_suid suid_write_to_sgid suid_write_to_suid_sgid
```

Comment:
Can you merge these 3 binaries together and add some minimal argument processing to allow a specific test to be selected.  Then a `.ksh` wrapper script can be used to invoke each test.  This is a little more consistent with the other tests and would let you get rid of the shared header which contains the bulk of the test case.

Reply:
Changed the .h to .c with main(argc, argv) which takes argv[1] as a test type from .ksh files.